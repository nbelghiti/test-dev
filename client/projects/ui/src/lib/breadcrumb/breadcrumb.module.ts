import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonModule } from '../button/button.module';
import { FilterModule } from '../filter/filter.module';
import { FlyoutModule } from '../flyout/flyout.module';
import { IconModule } from '../icon/icon.module';

import { BreadcrumbItemComponent } from './breadcrumb-item/breadcrumb-item.component';
import { BreadcrumbComponent } from './breadcrumb/breadcrumb.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		ButtonModule,
		RouterModule,
		FlyoutModule,
		FilterModule,
		ReactiveFormsModule,
		TranslateModule
	],
	declarations: [
		BreadcrumbItemComponent,
		BreadcrumbComponent,
	],
	exports: [
		BreadcrumbItemComponent,
		BreadcrumbComponent,
	],
})
export class BreadcrumbModule {}
