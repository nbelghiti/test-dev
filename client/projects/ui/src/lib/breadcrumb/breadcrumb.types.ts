import { IFilterMenuItem } from '../filter/filter-menu/filter-menu.component.types';

export interface IBreadcrumbItem {
	value: string;
	label?: string;
	link?: string[];
	queryParams?: { [key: string]: string };
	items?: IFilterMenuItem[];
}
