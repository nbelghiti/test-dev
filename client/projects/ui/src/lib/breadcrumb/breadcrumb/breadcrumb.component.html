<nav class="c-breadcrumb">
	<div class="c-breadcrumb__content">
		<ul
			class="c-breadcrumb__list"
			[ngClass]="{
				'c-breadcrumb--border': border
		}"
		>
			<li *ngFor="let item of items">
				<ui-breadcrumb-item
					[first]="first"
					[label]="item?.label"
					[value]="item?.value"
					[link]="item?.link"
					[items]="item?.items"
					[queryParams]="item?.queryParams"
				>
				</ui-breadcrumb-item>
			</li>
		</ul>

		<ui-button
			*ngIf="editLink"
			classes="u-margin-left-2x u-border-top-none u-border-bottom-none"
			[iconBefore]="IconName.edit"
			[link]="editLink"
			[type]="ButtonType.secondary"
		>
			{{ 'EDIT_COMPONENT__EDIT_FIELDS' | translate }}
		</ui-button>
	</div>
</nav>
