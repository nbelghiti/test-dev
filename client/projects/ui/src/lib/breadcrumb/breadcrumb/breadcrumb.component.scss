// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Breadcrumb
// =============================================================================
.c-breadcrumb {
	width: 100%;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-breadcrumb__list {
	@include extend-list-clear();

	display: flex;
}

.c-breadcrumb__content {
	display: flex;
	justify-content: space-between;
	margin: 0 auto;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-breadcrumb--border {
	border: $border-width-base solid $border-color-grey-86;
	border-right: none;
}
