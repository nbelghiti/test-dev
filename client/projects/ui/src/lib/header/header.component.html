<div class="c-header">
	<div class="c-header__container">
		<div class="c-header__item">
			<a href="/" class="c-header__link">
				<img
					class="c-header__image"
					src="/assets/logo/logo.svg"
					alt=""
				>

				<h2
					*ngIf="!backLink"
					class="c-header__title-container"
				>
					<span class="c-header__title">{{ 'APP_NAME' | translate }}</span>
					<span class="c-header__subtitle">{{ 'APP_SUBTITLE' | translate }}</span>
				</h2>
			</a>

			<ui-icon-button
				*ngIf="backLink"
				class="c-header__button"
				[link]="backLink"
				[fluid]="true"
				[border]="false"
			>
				<ui-icon [name]="IconName.arrowLeft"></ui-icon>
			</ui-icon-button>
		</div>

		<div class="c-header__item c-header__item--stretch">
			<ui-breadcrumb
				*ngIf="breadcrumb?.length"
				[items]="breadcrumb"
				[editLink]="editLink"
				[first]=true
			>
			</ui-breadcrumb>
		</div>

		<div class="c-header__item c-header__item-actions">
			<div
				#notifications
				class="c-header__action"
			>
				<ng-content select="ui-notifications"></ng-content>
			</div>

			<div class="c-header__action">
				<ng-content select="ui-search-indicator"></ng-content>
			</div>

			<div class="c-header__action">
				<ng-content select="ui-settings-indicator"></ng-content>
			</div>

			<ng-content select="ui-user-info"></ng-content>
		</div>
	</div>
</div>
