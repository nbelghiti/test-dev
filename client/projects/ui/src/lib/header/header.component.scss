// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Header
// =============================================================================
.c-header {
	display: flex;
	flex-direction: column;
	justify-content: space-between;

	border-bottom: $border-width-base solid $border-color-grey-76;

	background-color: $background-color-base;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-header__container {
	display: flex;
	justify-content: space-between;
	align-items: center;
}

.c-header__item {
	display: flex;
	align-items: center;
	flex-shrink: 0;
}

.c-header__item-actions {
	margin: 0 2.4rem 0 1rem;
	padding: 1rem 0;
}

.c-header__image {
	width: 4.4rem;
	height: 4.4rem;

	transform-origin: center;

	transition: $animation-base;
}

.c-header__title-container {
	margin-left: 1rem;

	font-family: $font-family-quaternary;
	font-size: $font-size-2x-large;

	color: $text-color-white;
}

.c-header__title {
	display: block;

	font-weight: $font-weight-dark;
	font-size: $font-size-2x-large;
	text-transform: lowercase;
	line-height: $line-height-normalize;
	letter-spacing: 0.1rem;
}

.c-header__subtitle {
	display: block;

	font-size: $font-size-2x-small;
	font-weight: $font-weight-normal;
	line-height: $line-height-normalize;
}

.c-header__action {

	&:not(:empty) {
		margin-right: 2rem;
	}
}

.c-header__link {
	display: flex;
	align-items: center;
	padding: 1rem 2.2rem;

	text-decoration: none;

	background:
		linear-gradient(
			135deg,
			$background-color-orange-54 0%,
			$background-color-orange-48 100%
		);

	&:hover {

		.c-header__image {
			transform: scale(1.1);
		}
	}
}

.c-header__button {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 5rem;
	// specific height is needed for the borders
	height: 6.4rem;

	border-right: $border-width-1x-large solid $border-color-base;

	transition: $animation-base;

	&:hover {
		color: $text-color-base;
	}
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-header__item--stretch {
	flex: 1;
}
