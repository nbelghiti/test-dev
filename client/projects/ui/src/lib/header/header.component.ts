import { Component, Input } from '@angular/core';

import { ButtonType } from '../button/button.types';
import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-header',
	templateUrl: './header.component.html',
	styleUrls: ['./header.component.scss']
})

export class HeaderComponent {
	@Input() public home?: boolean;
	@Input() public backLink?: string;
	@Input() public breadcrumb?: any;
	@Input() public editLink?: string[] = null;

	public ButtonType: typeof ButtonType = ButtonType;
	public IconName: typeof IconName = IconName;
}
