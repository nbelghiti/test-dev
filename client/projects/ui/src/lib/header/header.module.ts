import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { BreadcrumbModule } from '../breadcrumb/breadcrumb.module';
import { ButtonModule } from '../button/button.module';
import { IconModule } from '../icon/icon.module';

import { HeaderComponent } from './header.component';

@NgModule({
	imports: [
		CommonModule,
		RouterModule,
		IconModule,
		ButtonModule,
		BreadcrumbModule,
		TranslateModule,
	],
	declarations: [
		HeaderComponent,
	],
	exports: [
		HeaderComponent,
	],
})
export class HeaderModule { }
