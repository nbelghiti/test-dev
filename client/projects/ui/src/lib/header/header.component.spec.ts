import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { BreadcrumbModule } from '../breadcrumb/breadcrumb.module';
import { ButtonModule } from '../button/button.module';
import { IconModule } from '../icon/icon.module';

import { HeaderComponent } from './header.component';

describe('HeaderComponent', () => {
	let component: HeaderComponent;
	let fixture: ComponentFixture<HeaderComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				ReactiveFormsModule,
				TranslateModule.forRoot(),
				BreadcrumbModule,
				ButtonModule,
				IconModule,
			],
			declarations: [
				HeaderComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(HeaderComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});

	it('should not render a settings indicator when no ui-settings-indicator is provided', () => {
		expect(
			fixture.debugElement.query(By.css('ui-settings-indicator'))
		).toBeFalsy();
	});
});
