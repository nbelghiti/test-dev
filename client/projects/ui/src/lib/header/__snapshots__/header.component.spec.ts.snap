// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Should render correctly 1`] = `
<ui-header
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  editLink="null"
>
  <div
    class="c-header"
  >
    <div
      class="c-header__container"
    >
      <div
        class="c-header__item"
      >
        <a
          class="c-header__link"
          href="/"
        >
          <img
            alt=""
            class="c-header__image"
            src="/assets/logo/logo.svg"
          />

          <h2
            class="c-header__title-container"
          >
            <span
              class="c-header__title"
            >
              APP_NAME
            </span>
            <span
              class="c-header__subtitle"
            >
              APP_SUBTITLE
            </span>
          </h2>
        </a>

      </div>
      <div
        class="c-header__item c-header__item--stretch"
      >

      </div>
      <div
        class="c-header__item c-header__item-actions"
      >
        <div
          class="c-header__action"
        />
        <div
          class="c-header__action"
        />
        <div
          class="c-header__action"
        />
      </div>
    </div>
  </div>
</ui-header>
`;
